lib/Core/SpecialFunctionHandler.cpp

```diff
@@ -450,8 +454,52 @@ void SpecialFunctionHandler::handleMemalign(ExecutionState &state,
         0, "Symbolic alignment for memalign. Choosing smallest alignment");
   }
 
-  executor.executeAlloc(state, arguments[1], false, target, false, 0,
-                        alignment);
+  // Try to split up the memory allocation
```

Comment:
Okay, so, we have two different mechanisms to try to break up pmem allocations? First we try to split it into some MemoryObjects and then we use slice constraints as well? I feel like these two step on each others' toes. What are your thoughts?

Reply:
The slice constraints currently only return a single constraint. I was experimenting with the methodology---this current method is the most effective. I'll add a refactor TODO.

Reply:
Oh so we're ditching the slice constraints and just splitting up the memory objects?

I find that curious.. I feel like I would have had more faith in the slice constraints. But I don't know much about it. Could you elaborate on how you evaluated the effectiveness of each approach?